kvdb/src/lib.rs

```diff
@@ -20,6 +20,8 @@ pub const PREFIX_LEN: usize = 12;
 pub type DBValue = Vec<u8>;
 /// Database keys.
 pub type DBKey = SmallVec<[u8; 32]>;
+/// A tuple holding key and value data, used in the iterator item type.
+pub type KeyValuePair = (Box<[u8]>, Box<[u8]>);
```

Comment:
There's a small discrepancy with value here being `Box<[u8]>` while `DBValue` is `Vec<u8>`. Is it worth fixing? Not sure.

Reply:
I admit I would be prone to say makes it all Vec<u8> (I don't think to_vec call on box<[u8]> is expensive).
But that's just my opinion.

Reply:
`Box -> Vec` is cheap indeed, I'll change it to `(DBKey, DBValue)`.